src/app/Handlers/Json/Updater.php

```diff
@@ -21,6 +21,11 @@ public function run()
         $this->processDifferences();
     }
 
+    public function add()
+    {
+        $this->processDifferences();
+    }
+
```

Comment:
I thought the same but I tested it and it has the desired behaviour - only new keys are added.

Reply:
This would mean that `removeExtraKeys` is not working properly.

Can you `\Log::info` the `$keysToRemove` after 
```
$keysToRemove = collect($extraLangFile)
            ->diffKeys($this->updatedLangFile);
```
and share the response?

Reply:
will check this asap.

Reply:
there are 2 problems:

1st: $keyToRemove in Loop is mapped to value, not to key
2nd: $extraLangFile is not updated

New function (tested):
``
    private function removeExtraKeys(array $extraLangFile)
    {
        $keysToRemove = collect($extraLangFile)
            ->diffKeys($this->updatedLangFile);

        $keysToRemove->each(function ($valueToRemove, $keyToRemove) use (&$extraLangFile) {
            unset($extraLangFile[$keyToRemove]);
        });

        return [$keysToRemove->count(), $extraLangFile];
    }
``